server/models/User.coffee

```diff
@@ -264,12 +264,7 @@ UserSchema.statics.unconflictName = unconflictName = (name, done) ->
 
 UserSchema.methods.register = (done) ->
   @set('anonymous', false)
-  if (name = @get 'name')? and name isnt ''
-    unconflictName name, (err, uniqueName) =>
-      return done err if err
-      @set 'name', uniqueName
-      done()
-  else done()
+  done()
```

Comment:
No, all `done` did before was provide a way for the function calling `register` to wait for the name to be unconflicted if it needed to be unconflicted. Since that no longer needs to be waited on by the caller, and the rest we still don't want to blocking, there's no need for a callback to provide a mechanism for the caller to be blocked.